astropy/nddata/bitmask.py

```diff
@@ -0,0 +1,411 @@
+"""
+A module that provides functions for manipulating bitmasks and data quality
+(DQ) arrays.
+
+"""
+
+import sys
+import warnings
+import numpy as np
+
+
+__all__ = ['bitfield_to_boolean_mask', 'interpret_bit_flags', 'is_bit_flag']
+
+
+INT_TYPE = (int, long,) if sys.version_info < (3,) else (int,)
+MAX_UINT_TYPE = np.maximum_sctype(np.uint)
+SUPPORTED_FLAGS = int(np.bitwise_not(
+    0, dtype=MAX_UINT_TYPE, casting='unsafe'
+))
+
+
+def is_bit_flag(n):
```

Comment:
Would you object to turning this into a "private" function `_is_bit_flag`? With the caveat that I haven't used bit flags extensively it seems like this is most useful internally.

Reply:
@mwcraig I was actually thinking the same thing a couple of hours earlier but I simply forgot about it. I also do not see any obvious use for this function (unless somebody would use it for development - not for direct use by users). It can always be made public later if there enough interest. It is more difficult to make it private later.

Probably I have developed this function to be used instead of https://github.com/MSeifert04/ccdproc/blob/81eea65e161168af2edd4d2acd8770623eacc53a/ccdproc/bitmask.py#L14 My original code did not have an extensive validation of inputs. I have a feeling a developer might benefit from this function if they will ever want to implement something like ["name maps"](https://github.com/MSeifert04/ccdproc/blob/81eea65e161168af2edd4d2acd8770623eacc53a/ccdproc/bitmask.py#L14) (also see "maps of flag names" in https://github.com/astropy/astropy/issues/4741#issue-143825905). So, if anyone developing something on top of `bitmask` would like to validate bit flags (provided as input or as values in a dict/map), this function might be useful. I will need to look at our own code here if I use it anywhere outside `bitmask`. If not - I am fine with making this function private (on the other hand making it visible will help avoiding people going through something like https://github.com/MSeifert04/ccdproc/blob/81eea65e161168af2edd4d2acd8770623eacc53a/ccdproc/bitmask.py#L14).

Reply:
Yep, I do not see this function being used anywhere else outside `bitmask`. @mwcraig Let me know if you do not think this function might be of use to _developers_. If you do - I'd say let's leave it public (???)

Reply:
I'd lean towards private since more dev types will probably dig into the source code anyway.